src/Jonathan_Requirements/Jonathan_Requirements.c

```diff
@@ -0,0 +1,51 @@
+// Requirements.cpp
+// Created by Jonathan Morelock 3/12/2021
+#include <math.h>
+#include "Jonathan_Requirements.h"
+
+int set_draw_distance(int *render_radius, float speed, Precipitation * P)
+{
+    if (speed < 5 || P->type == RAIN)
+    {
+        *render_radius = 5;
+    }
+    if (speed >= 5 && speed <= 10 || P->type == SNOW)
+    {
+        //Draw slow rate
+        *render_radius = 10;
+    }
+    else if (speed > 10 && speed <= 15 && P->type == CLEAR)
+    {
+        //Draw medium rate
+        *render_radius = 17;
+    }
+    else if (speed > 15 && P->type == CLEAR)
+    {
+        //Draw fast rate
+        *render_radius = 24; //max render radius as set on line 2068 in main.c
+    }
+    return *render_radius;
+}
+
+void generate_precipitation_vectors(Precipitation *P, float x, float y, float z)
+{
+    P->pv.x = x;
+    P->pv.y = y;
+    P->pv.z = z;
+}
+
+void set_precipitation_type(Precipitation *P, int precip_type)
```

Comment:
I don't know why you need this function because your header file `typedef enum
{
    RAIN,
    SNOW,
    CLEAR
} PrecipTypes;`
which set RAIN=0, SNOW =1, CLEAR=2;

Reply:
I am using it in the test bench, I generate a random int 0 to 2 and pass it to this function to select which precipitation will be used over a chunk.  I am going to update the test bench to dynamically select the size of the random int just in case we add more weather conditions soon. If there is a better way to do it let me know! Thanks!